src/mito2/src/region.rs

```diff
@@ -42,33 +50,18 @@ pub(crate) struct RegionMap {
     regions: RwLock<HashMap<RegionId, MitoRegionRef>>,
 }
 
-pub(crate) type RegionMapRef = Arc<RegionMap>;
-
-/// [MitoRegion] builder.
-pub(crate) struct RegionBuilder {
-    metadata: RegionMetadataRef,
-    memtable_builder: MemtableBuilderRef,
-}
-
-impl RegionBuilder {
-    /// Returns a new builder.
-    pub(crate) fn new(
-        metadata: RegionMetadataRef,
-        memtable_builder: MemtableBuilderRef,
-    ) -> RegionBuilder {
-        RegionBuilder {
-            metadata,
-            memtable_builder,
-        }
+impl RegionMap {
```

Comment:
FWIW this struct should be cross-engine

Reply:
Currently, I prefer to keep each worker's region map separated to avoid workers modifying other workers' regions by mistake.